tools/clang/include/clang/SPIRV/ModuleBuilder.h

```diff
@@ -93,7 +103,23 @@ void ModuleBuilder::requireCapability(spv::Capability cap) {
   theModule.addCapability(cap);
 }
 
+void ModuleBuilder::addEntryPoint(spv::ExecutionModel em, uint32_t targetId,
+                                  std::string targetName,
+                                  std::initializer_list<uint32_t> interfaces) {
+  theModule.addEntryPoint(em, targetId, targetName, interfaces);
+}
+
+void ModuleBuilder::addExecutionMode(uint32_t entryPointId,
```

Comment:
This method should be moved into `.cpp`.

Add `.x()` should happen at the very last, after all additional parameters are supplied:
```cpp
instBuilder.opExecutionMode(entryPointId, em);
for (...) { ... }
instBuilder.x();
...
```